src/druglab/db/backend/memory.py

```
@@ -688,11 +735,24 @@ def load(
         # --- objects ---
         obj_path = path / "objects" / "objects.pkl"
         if obj_path.exists():
-            raw_list = pickle.loads(obj_path.read_bytes()) 
-            if deserializer is not None:
-                objects = [deserializer(r) for r in raw_list]
-            else:
-                objects = raw_list
+            with open(obj_path, "rb") as f:
+                raw_payload = pickle.load(f)
```

Ruff (0.15.10)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)